src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2881,6 +2881,16 @@ private bool isFieldStatic(CORINFO_FIELD_STRUCT_* fldHnd)
             return HandleToObject(fldHnd).IsStatic;
         }
 
+#pragma warning disable CA1822 // Mark members as static
+        private int getReadonlyStaticArrayLength(CORINFO_FIELD_STRUCT_* fldHnd)
+#pragma warning restore CA1822 // Mark members as static
+        {
+            // This is not used on crossgen and NativeAOT.
+            // For NativeAOT we'll need a different API, e.g. "getReadonlyStaticArrayLength(void* frozenObjPtr)"
```

Comment:
This should be addressed if you take my other suggestion. Frozen objects are natural handles.